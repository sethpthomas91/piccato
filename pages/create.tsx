import React from 'react'
import dynamic from 'next/dynamic'

const Ml5 = dynamic(() => import('../components/Ml5'), {ssr: false,})

const Create: React.FC = () => {
  return (
    <Ml5/>
  );
}
  
export default Create;
  