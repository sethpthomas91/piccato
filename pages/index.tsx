import styles from '../styles/gallery.module.css'
import { GetServerSideProps } from 'next'
import GalleryImage, { GalleryImageProps } from '../components/GalleryImage'

type Props = {
  data: GalleryImageProps[];
}

export const generateImageArr = (props:Props) => {
  return (props.data.map(image => (
    <GalleryImage key={image.id} image={image}/>
  )))
}

export const getServerSideProps: GetServerSideProps = async () => {
  const response = await fetch('http://localhost:3000/api/images/');
  const data = await response.json();
  return { 
    props: { data },
  };
}

const Home: React.FC<Props> = (props) => {
  return (
    <>
      <main className={styles.gallery}>
        {generateImageArr(props)}
      </main>
    </>
  )
}

export default Home
