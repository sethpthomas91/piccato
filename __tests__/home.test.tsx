import '@testing-library/jest-dom';
import { generateImageArr } from '../pages/index';
import Home from '../pages/index';
import renderer from 'react-test-renderer';

describe ('generateImageArr', () => {
  const mockImageData = {
    data: [
      {
        name: "This does not matter",
        url: "some website",
        createdAt: "Some time",
        id: 1
      },
      {
        name: "This does not matter",
        url: "some website",
        createdAt: "Some time",
        id: 2
      },
      {
        name: "This does not matter",
        url: "some website",
        createdAt: "Some time",
        id: 3
      }
    ]
  };

  it('returns an array of 3 images', () => {
    const imageArr = generateImageArr(mockImageData);
    expect(imageArr).toHaveLength(3)
  });
})

describe ('Home', () => {
  const mockImageData = {
    data: [
      {
        name: "This does not matter",
        url: "some website",
        createdAt: "Some time",
        id: 1
      },
      {
        name: "This does not matter",
        url: "some website",
        createdAt: "Some time",
        id: 2
      },
      {
        name: "This does not matter",
        url: "some website",
        createdAt: "Some time",
        id: 3
      }
    ]
  };

  it('renders correctly', () => {
    const tree = renderer
      .create(<Home data={mockImageData.data}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})
