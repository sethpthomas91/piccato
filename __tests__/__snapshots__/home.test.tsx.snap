// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<main
  className="gallery"
>
  <div
    className="container"
    onClick={[Function]}
  >
    <img
      className="galleryImg"
      src="some website"
    />
  </div>
  <div
    className="container"
    onClick={[Function]}
  >
    <img
      className="galleryImg"
      src="some website"
    />
  </div>
  <div
    className="container"
    onClick={[Function]}
  >
    <img
      className="galleryImg"
      src="some website"
    />
  </div>
</main>
`;
